pysrc/bytewax/inputs.py

```diff
@@ -90,6 +91,26 @@ def close(self) -> None:
         """
         pass
 
+    def next_awake(self) -> datetime:
+        """Return the time for the next poll of the input.
+        By default this is `now`, which means the input will
+        be polled continously.
+
+        This function is called before `self.next`, and if the
```

Comment:
It used to be that way, but I changed it so I could skip `next` calls during spurious wake-ups. Setting `activate_after` doesn't guarantee that the operator will be called after that amount of time, so I often (though not always) saw additional `next` calls immediately after the first one from Timely. I couldn't find a way to prevent this, so as a workaround, I decided to check `next_awake` every time and explicitly avoid calling `next` if the next wake-up time is in the future. This made the PeriodicInput behave as expected, while before it was producing a variable number of values per period. The number of wake ups seemed mostly consistent during a run, but changed between runs (sometimes 1 value, sometimes 2, sometimes 3)